includes/data-stores/class-wc-admin-reports-categories-data-store.php

```diff
@@ -148,6 +148,22 @@ protected function page_records( $data, $page_no, $items_per_page ) {
 		return array_slice( $data, $offset, $items_per_page );
 	}
 
+	/**
+	 * Enriches the category data.
+	 *
+	 * @param array $categories_data Categories data.
+	 * @param array $query_args  Query parameters.
+	 */
+	protected function include_extended_info( &$categories_data, $query_args ) {
+		foreach ( $categories_data as $key => $category_data ) {
+			$extended_info = new ArrayObject();
+			if ( $query_args['extended_info'] ) {
+				$extended_info['name'] = get_the_category_by_ID( $category_data['category_id'] );
```

Comment:
I _think_ this is okay from a performance perspective but would want to get @peterfabian / @claudiosanches input on it. First inclination for me was that we should grab a mapping of all category ID=> names and reference that mapping while looping `$categories_data`

Reply:
A mapping would be useful an API driven approach for nested categories (https://github.com/woocommerce/wc-admin/issues/1045), but we'd need to do that for the Products tabular data as well.

`get_the_category_by_ID` uses [get_term](https://core.trac.wordpress.org/browser/tags/5.0/src/wp-includes/taxonomy.php#L754), but I'm not sure either of the performance implications.

Reply:
ping @peterfabian @claudiosanches